test-scripts/load-fixtures.js

```
@@ -2,80 +2,139 @@ const _ = require("lodash");
-const actorOrg = require("../test/fixtures/organizations/org-good-health.json");
-const consent = require(CONSENT_FILE);
-
-patient.id = "thePatient";
-consentOrg.id = "theConsentOrg";
-actorOrg.id = "theActorOrg";
-consent.patient.reference = `Patient/${patient.id}`;
-consent.organization[0].reference = `Organization/${consentOrg.id}`;
-consent.provision.provision = consent.provision.provision.map(provision =>
-  _.set(
-    _.cloneDeep(provision),
-    "actor[0].reference.reference",
-    `Organization/${actorOrg.id}`
-  )
-);
-consent.dateTime = new Date().toISOString();
-
-const entries = [patient, actorOrg, consentOrg, consent].map(resource => ({
-  fullUrl: "",
-  resource: resource,
-  request: {
-    method: "PUT",
-    url: `${
-      resource.resourceType
-    }?identifier=${LEAP_IDENTIFIER_SYSTEM}|${getLeapIdentifer(resource)}`
+async function removeResourcesByTypes(resourceTypes) {
+  for (const resourceType of resourceTypes) {
+    const ids = await getAllResourceIds(resourceType);
+    await removeAllResourcesByTypeAndIds(resourceType, ids);
   }
-}));
-
-const transaction = {
-  resourceType: "Bundle",
-  id: "bundle-transaction",
-  meta: {
-    lastUpdated: new Date().toISOString()
-  },
-  type: "transaction",
-  entry: entries
-};
-
-superagent
-  .post(FHIR_BASE)
-  .set("Content-Type", "application/json")
-  .send(transaction)
-  .then(res => {
-    console.log(JSON.stringify(res.body, null, 2));
-  })
-  .catch(err => {
-    console.log(err);
-  });
+}
+
+function prepareEntries() {
+  const consent = require(`${PATH}/consents/r5/${CONSENT_FILE}`);
+  const patient = require(`${PATH}/patients/patient-boris.json`);
+  const consentOrg = require(`${PATH}/organizations/org-hl7.json`);
+  const actorOrg = require(`${PATH}/organizations/org-good-health.json`);
+
+  patient.id = "thePatient";
+  consentOrg.id = "theConsentOrg";
+  actorOrg.id = "theActorOrg";
+
+  consent?.subject?.reference &&
```

I think it's best if these are not conditional so that they fail if the resource is malformed instead of silently pushing a resource without correct linking